### PR TITLE
fix[Scripting]: Scripting Debug

### PR DESCRIPTION
**Pull request by ericdacal**
Now the scripts work again

### CODE REVIEW

File: Game/Scripts/ActivationLogic.cpp [hunk 1]
```diff
@@ -53,21 +57,26 @@ void ActivationLogic::Update(float deltaTime)
...
 {
-	if (other->GetOwner()->GetComponent<ComponentPlayer>())
-	{
+	try {
+		other->GetOwner()->GetComponent<ComponentPlayer>();
 		componentAnimation->SetParameter("IsActive", true);
 		componentRigidBody->Disable();
 		componentAudio->PostEvent(AUDIO::SFX::AMBIENT::SEWERS::BIGDOOR_OPEN);
 	}
+	catch(const ComponentNotFoundException){
```
Comment (gerard-queralt) [inline Game/Scripts/ActivationLogic.cpp:66]: From what I gathered, the C++ standard is to catch by reference. Please change this catch clause and the one in line 80.
```suggestion
	catch(const ComponentNotFoundException&){
```

File: Game/Scripts/TriggerSewersMusic.cpp [hunk 2]
```diff
@@ -33,4 +35,8 @@ void TriggerSewersMusic::OnCollisionEnter(ComponentRigidBody* other)
 			isMusicTriggered = true;
 		}
 	}
+	catch (const ComponentNotFoundException&)
+	{
+		LOG_WARNING("{} have not ComponentPlayer", other->GetOwner()->GetName());
```
Comment (gerard-queralt) [inline Game/Scripts/TriggerSewersMusic.cpp:40]: Should this have a log? If I'm reading it correctly, this was just a check, and we're not storing the `ComponentPlayer` anywhere. I guess the question is: is this something unexpected that should not happen? Same goes for the other instances where this check (`GetComponent<ComponentPlayer>() != nullptr`) was previously performed.

File: Game/Scripts/TriggerSewersMusic.cpp [hunk 3]
```diff
@@ -21,8 +22,9 @@ void TriggerSewersMusic::Start()
 
 void TriggerSewersMusic::OnCollisionEnter(ComponentRigidBody* other)
 {
-	if (other->GetOwner()->GetComponent<ComponentPlayer>())
-	{
+
+	try {
+		other->GetOwner()->GetComponent<ComponentPlayer>();
```
Comment (DADSILVER) [inline Game/Scripts/TriggerSewersMusic.cpp:27]: This should be GetTag() == "Player" as the others.

File: Game/Scripts/TriggerSewersMusicExit.cpp [hunk 4]
```diff
@@ -26,12 +27,16 @@ void TriggerSewersMusicExit::Start()
 
 void TriggerSewersMusicExit::OnCollisionEnter(ComponentRigidBody* other)
 {
-	if (other->GetOwner()->GetComponent<ComponentPlayer>())
-	{
+	try {
+		other->GetOwner()->GetComponent<ComponentPlayer>();
```
Comment (DADSILVER) [inline Game/Scripts/TriggerSewersMusicExit.cpp:31]: This should be GetTag() == "Player" as the others.

File: Game/Scripts/UIMissionTrigger.cpp [hunk 5]
```diff
@@ -58,8 +65,8 @@ void UIMissionTrigger::Update(float deltaTime)
 
 void UIMissionTrigger::OnCollisionEnter(ComponentRigidBody* other)
 {
-	if (other->GetOwner()->GetComponent<ComponentPlayer>())
-	{
+	try {
+		other->GetOwner()->GetComponent<ComponentPlayer>();
```
Comment (DADSILVER) [inline Game/Scripts/UIMissionTrigger.cpp:69]: This should be GetTag() == "Player" as the others.

File: Game/Scripts/UITrigger.cpp [hunk 6]
```diff
@@ -89,16 +90,24 @@ void UITrigger::Update(float deltaTime)
 
 void UITrigger::OnCollisionEnter(ComponentRigidBody* other)
 {
-	if (other->GetOwner()->GetComponent<ComponentPlayer>())
-	{
+	try {
+		other->GetOwner()->GetComponent<ComponentPlayer>();
```
Comment (DADSILVER) [inline Game/Scripts/UITrigger.cpp:94]: This should be GetTag() == "Player" as the others.

File: Game/Scripts/UITrigger.cpp [hunk 7]
```diff
@@ -89,16 +90,24 @@ void UITrigger::Update(float deltaTime)
...
+		LOG_WARNING("{} have not ComponentPlayer", other->GetOwner()->GetName());
+	}
 }
 
 void UITrigger::OnCollisionExit(ComponentRigidBody* other)
 {
-	if (other->GetOwner()->GetComponent<ComponentPlayer>())
-	{
+	try {
+		other->GetOwner()->GetComponent<ComponentPlayer>();
```
Comment (DADSILVER) [inline Game/Scripts/UITrigger.cpp:106]: This should be GetTag() == "Player" as the others.

**Review by DADSILVER — DISMISSED** [review]
Thx a lot for this PR. Only 1 question and 1 thing. Should the assets be added to the PR?

The other thing is that we should use GetTag instead of get component if possible. GetComponente is most costly than GetTag.

File: Game/Scripts/ActivationLogic.cpp [hunk 8]
```diff
@@ -36,10 +36,12 @@ void ActivationLogic::Start()
 	componentRigidBody = nullptr;
 	for (GameObject* child : owner->GetChildren())
 	{
-		componentRigidBody = child->GetComponent<ComponentRigidBody>();
-		if (componentRigidBody != nullptr)
-		{
-			break;
+		try {
```
Comment (markmeliaa) [inline Game/Scripts/ActivationLogic.cpp:39]: The opening bracket of this `try...catch` block should go on the next line, as stated in our programming rules.

File: Game/Scripts/ActivationLogic.cpp [hunk 9]
```diff
@@ -36,10 +36,12 @@ void ActivationLogic::Start()
...
+			componentRigidBody = child->GetComponent<ComponentRigidBody>();
```
Comment (markmeliaa) [inline Game/Scripts/ActivationLogic.cpp:40]: Before, when a `ComponentRigidBody` was found, there was a `break`, should it be added here as well?

File: Game/Scripts/CantinaNPCAnimationManager.cpp [hunk 10]
```diff
@@ -13,7 +14,13 @@ CantinaNPCAnimationManager::CantinaNPCAnimationManager() : Script(), audio(nullp
 
 void CantinaNPCAnimationManager::Start()
 {
-	audio = owner->GetComponent<ComponentAudioSource>();
+	try {
```
Comment (markmeliaa) [inline Game/Scripts/CantinaNPCAnimationManager.cpp:17]: The opening bracket of this `try...catch` block should go on the next line, as stated in our programming rules.

File: Game/Scripts/CantinaNPCAnimationManager.cpp [hunk 11]
```diff
@@ -13,7 +14,13 @@ CantinaNPCAnimationManager::CantinaNPCAnimationManager() : Script(), audio(nullp
...
+		audio = owner->GetComponent<ComponentAudioSource>();
+	}
+	catch (const ComponentNotFoundException&)
+	{
+		LOG_WARNING("{} have not Component Audio Source", owner->GetName());
```
Comment (markmeliaa) [inline Game/Scripts/CantinaNPCAnimationManager.cpp:22]:
```suggestion
		LOG_WARNING("{} has not Component Audio Source", owner->GetName());
```

File: Game/Scripts/ActivationLogic.cpp [hunk 12]
```diff
@@ -53,18 +55,18 @@ void ActivationLogic::Update(float deltaTime)
...
 		componentAudio->PostEvent(AUDIO::SFX::AMBIENT::SEWERS::BIGDOOR_OPEN);
-	}
+	}	
 }
 
 void ActivationLogic::OnCollisionExit(ComponentRigidBody* other)
 {
-	if (other->GetOwner()->GetComponent<ComponentPlayer>())
-	{
+	LOG_VERBOSE("{} enter in CollisionExit",other->GetOwner()->GetName());
```
Comment (markmeliaa) [inline Game/Scripts/ActivationLogic.cpp:68]:
```suggestion
	LOG_VERBOSE("{} enters in CollisionExit",other->GetOwner()->GetName());
```

File: Game/Scripts/ActivationLogic.cpp [hunk 13]
```diff
@@ -53,18 +55,18 @@ void ActivationLogic::Update(float deltaTime)
 
 void ActivationLogic::OnCollisionEnter(ComponentRigidBody* other)
 {
-	if (other->GetOwner()->GetComponent<ComponentPlayer>())
-	{
+	LOG_VERBOSE("{} enter in CollisionEnter", other->GetOwner()->GetName());
+	if (other->GetOwner()->GetTag() == "Player") {
```
Comment (markmeliaa) [inline Game/Scripts/ActivationLogic.cpp:59]:
```suggestion
	if (other->GetOwner()->CompareTag("Player")) {
```

File: Game/Scripts/ActivationLogic.cpp [hunk 14]
```diff
@@ -53,18 +55,18 @@ void ActivationLogic::Update(float deltaTime)
...
-	}
+	}	
 }
 
 void ActivationLogic::OnCollisionExit(ComponentRigidBody* other)
 {
-	if (other->GetOwner()->GetComponent<ComponentPlayer>())
-	{
+	LOG_VERBOSE("{} enter in CollisionExit",other->GetOwner()->GetName());
+	if (other->GetOwner()->GetTag() == "Player") {
```
Comment (markmeliaa) [inline Game/Scripts/ActivationLogic.cpp:69]:
```suggestion
	if (other->GetOwner()->CompareTag("Player")) {
```

File: Game/Scripts/ActivationLogic.cpp (hunk 13 above)
Comment (markmeliaa) [reply Game/Scripts/ActivationLogic.cpp:59]: Please, when performing tag checks, use `CompareTag`.

File: Game/Scripts/HealthSystem.cpp [hunk 15]
```diff
@@ -24,7 +25,13 @@ void HealthSystem::Start()
 {
 	componentAnimation = owner->GetComponent<ComponentAnimation>();
 	//componentParticleSystem = enemyParticleSystem->GetComponent<ComponentParticleSystem>();
-	componentParticleSystem = owner->GetComponent<ComponentParticleSystem>();
+	try {
+		componentParticleSystem = owner->GetComponent<ComponentParticleSystem>();
+	}
+	catch (const ComponentNotFoundException&) {
+		LOG_WARNING("{} not have ComponentParticleSystem", owner->GetName());
```
Comment (markmeliaa) [inline Game/Scripts/HealthSystem.cpp:32]:
```suggestion
		LOG_WARNING("{} does not have ComponentParticleSystem", owner->GetName());
```

File: Game/Scripts/HealthSystem.cpp [hunk 16]
```diff
@@ -24,7 +25,13 @@ void HealthSystem::Start()
 {
 	componentAnimation = owner->GetComponent<ComponentAnimation>();
 	//componentParticleSystem = enemyParticleSystem->GetComponent<ComponentParticleSystem>();
-	componentParticleSystem = owner->GetComponent<ComponentParticleSystem>();
+	try {
```
Comment (markmeliaa) [inline Game/Scripts/HealthSystem.cpp:28]: The opening bracket of this `try...catch` block should go on the next line, as stated in our programming rules.

File: Game/Scripts/HealthSystem.cpp [hunk 17]
```diff
@@ -81,11 +88,16 @@ void HealthSystem::TakeDamage(float damage)
 		}
 
 		componentAnimation->SetParameter("IsTakingDamage", true);
-
-		if (componentParticleSystem)
-		{
-			componentParticleSystem->Play();
+		try {
```
Comment (markmeliaa) [inline Game/Scripts/HealthSystem.cpp:91]: The opening bracket of this `try...catch` block should go on the next line, as stated in our programming rules.

File: Game/Scripts/HealthSystem.cpp [hunk 18]
```diff
@@ -81,11 +88,16 @@ void HealthSystem::TakeDamage(float damage)
...
+			if (componentParticleSystem)
+			{
+				componentParticleSystem->Play();
+			}
+		}
+		catch (const ComponentNotFoundException&) {
```
Comment (markmeliaa) [inline Game/Scripts/HealthSystem.cpp:97]: The opening bracket of this `try...catch` block should go on the next line, as stated in our programming rules.

File: Game/Scripts/HealthSystem.cpp [hunk 19]
```diff
@@ -24,7 +25,13 @@ void HealthSystem::Start()
 {
 	componentAnimation = owner->GetComponent<ComponentAnimation>();
 	//componentParticleSystem = enemyParticleSystem->GetComponent<ComponentParticleSystem>();
-	componentParticleSystem = owner->GetComponent<ComponentParticleSystem>();
+	try {
+		componentParticleSystem = owner->GetComponent<ComponentParticleSystem>();
+	}
+	catch (const ComponentNotFoundException&) {
```
Comment (markmeliaa) [inline Game/Scripts/HealthSystem.cpp:31]: The opening bracket of this `try...catch` block should go on the next line, as stated in our programming rules.

File: Game/Scripts/ActivationLogic.cpp [hunk 20]
```diff
@@ -36,10 +36,12 @@ void ActivationLogic::Start()
...
 	{
-		componentRigidBody = child->GetComponent<ComponentRigidBody>();
-		if (componentRigidBody != nullptr)
-		{
-			break;
+		try {
+			componentRigidBody = child->GetComponent<ComponentRigidBody>();
+			LOG_VERBOSE("{} has Component Rigid Body", child->GetName());
+		}
+		catch (const ComponentNotFoundException&) {
```
Comment (markmeliaa) [inline Game/Scripts/ActivationLogic.cpp:43]: The opening bracket of this `try...catch` block should go on the next line, as stated in our programming rules.

File: Game/Scripts/ActivationLogic.cpp (hunk 14 above)
Comment (markmeliaa) [inline Game/Scripts/ActivationLogic.cpp:69]: The opening bracket of this `if statement` should go on the next line, as stated in our programming rules.

File: Game/Scripts/TriggerSewersMusic.cpp [hunk 21]
```diff
@@ -21,8 +22,9 @@ void TriggerSewersMusic::Start()
 
 void TriggerSewersMusic::OnCollisionEnter(ComponentRigidBody* other)
 {
-	if (other->GetOwner()->GetComponent<ComponentPlayer>())
-	{
+
+	try {
```
Comment (markmeliaa) [inline Game/Scripts/TriggerSewersMusic.cpp:26]: `try...catch` block not needed.

File: Game/Scripts/TriggerSewersMusic.cpp (hunk 3 above)
Comment (markmeliaa) [reply Game/Scripts/TriggerSewersMusic.cpp:27]: CompareTag("Player") instead, please. Same with all the following ones.

Also, changing this to a CompareTag eliminates the need of having the `try...catch` block.

File: Game/Scripts/TriggerSewersMusic.cpp (hunk 2 above)
Comment (markmeliaa) [reply Game/Scripts/TriggerSewersMusic.cpp:40]: IMO, I don't see the logs of the `try...catch` blocks necessary at all, its just "useless" info.

File: Game/Scripts/TriggerSewersMusicExit.cpp [hunk 22]
```diff
@@ -26,12 +27,16 @@ void TriggerSewersMusicExit::Start()
 
 void TriggerSewersMusicExit::OnCollisionEnter(ComponentRigidBody* other)
 {
-	if (other->GetOwner()->GetComponent<ComponentPlayer>())
-	{
+	try {
```
Comment (markmeliaa) [inline Game/Scripts/TriggerSewersMusicExit.cpp:30]: `try...catch` block not needed.

File: Game/Scripts/CantinaNPCAnimationManager.cpp (hunk 10 above)
Comment (markmeliaa) [reply Game/Scripts/CantinaNPCAnimationManager.cpp:17]: Is this `try...catch` needed? Would there be situations in which a `ComponentAudioSource` is not found?

File: Game/Scripts/UIMissionTrigger.cpp [hunk 23]
```diff
@@ -30,7 +30,14 @@ UIMissionTrigger::~UIMissionTrigger()
 
 void UIMissionTrigger::Start()
 {
-	player = App->GetModule<ModulePlayer>()->GetPlayer()->GetComponent<ComponentPlayer>();
+	try {
+		player = App->GetModule<ModulePlayer>()->GetPlayer()->GetComponent<ComponentPlayer>();
```
Comment (markmeliaa) [inline Game/Scripts/UIMissionTrigger.cpp:34]: Is this `try...catch` block needed? A player would always have a `ComponentPlayer`.

File: Game/Scripts/UIMissionTrigger.cpp [hunk 24]
```diff
@@ -58,8 +65,8 @@ void UIMissionTrigger::Update(float deltaTime)
 
 void UIMissionTrigger::OnCollisionEnter(ComponentRigidBody* other)
 {
-	if (other->GetOwner()->GetComponent<ComponentPlayer>())
-	{
+	try {
```
Comment (markmeliaa) [inline Game/Scripts/UIMissionTrigger.cpp:68]: Unnecessary `try...catch` block.

File: Game/Scripts/UITrigger.cpp [hunk 25]
```diff
@@ -89,16 +90,24 @@ void UITrigger::Update(float deltaTime)
 
 void UITrigger::OnCollisionEnter(ComponentRigidBody* other)
 {
-	if (other->GetOwner()->GetComponent<ComponentPlayer>())
-	{
+	try {
```
Comment (markmeliaa) [inline Game/Scripts/UITrigger.cpp:93]: Unnecessary `try...catch` block.

File: Game/Scripts/UITrigger.cpp [hunk 26]
```diff
@@ -89,16 +90,24 @@ void UITrigger::Update(float deltaTime)
...
+	{
+		LOG_WARNING("{} have not ComponentPlayer", other->GetOwner()->GetName());
+	}
 }
 
 void UITrigger::OnCollisionExit(ComponentRigidBody* other)
 {
-	if (other->GetOwner()->GetComponent<ComponentPlayer>())
-	{
+	try {
```
Comment (markmeliaa) [inline Game/Scripts/UITrigger.cpp:105]: Unnecessary `try...catch` block.

**Review by markmeliaa — CHANGES_REQUESTED** [review]
Good job Eric, thanks for fixing this., just a couple things.

If you did not change any model or any texture, the assets modified should not be uploaded, so if you could delete those, that would be nice!

File: Game/Scripts/TriggerSewersMusic.cpp (hunk 2 above)
Comment (gerard-queralt) [reply Game/Scripts/TriggerSewersMusic.cpp:40]: I think they're fine as long as they are temporary. Just to indicate that a given component should be there, for some reason isn't, but for now we leave it as is. But I wouldn't want them in the final version either. That's why I'm not sure these fit that criteria. Having said that, I trust your judgement over mine on this.

Comment (gerard-queralt) [reply Game/Scripts/TriggerSewersMusic.cpp:40]: I think they're fine as long as they are temporary. Just to indicate that a given component should be there, for some reason isn't, but for now we leave it as is. But I wouldn't want them in the final version either. That's why I'm not sure these fit that criteria. Having said that, I trust your judgement over mine on this.

**Review by Dedado3000 — APPROVED** [review]

**Comment by gerard-queralt** [comment]
Now that #616 has been merged, we could take a look at which `try...catch` blocks can be removed, starting with the one in `ActivationLogic::Start`.

File: Game/Assets/Materials/Lv1_CargoBay_RunWay2Body_0.mat [hunk 27]
```diff
@@ -1,5 +1,5 @@
 {
-    "DiffuseAssetPath": "Assets/Textures/Lv1_CargoBay_RunWay_Base.png",
+    "DiffuseAssetPath": "C:\\Users\\eric\\Desktop\\Pasarelas\\Texturas\\Pasarela_Pasarela_BaseColor.png",
```
Comment (MarcGonzalezMoratona) [inline Game/Assets/Materials/Lv1_CargoBay_RunWay2Body_0.mat:2]: Is this working?

File: Game/Assets/Materials/Lv1_CargoBay_RunWay2Barandilla_0.mat [hunk 28]
```diff
@@ -1,5 +1,5 @@
 {
-    "DiffuseAssetPath": "Assets/Textures/substanceBarandilla_Barandilla.002_BaseColor.png",
+    "DiffuseAssetPath": "C:\\Users\\eric\\Desktop\\Pasarelas\\Texturas\\substanceBarandilla_Barandilla.002_BaseColor.png",
```
Comment (MarcGonzalezMoratona) [inline Game/Assets/Materials/Lv1_CargoBay_RunWay2Barandilla_0.mat:2]: I think this path is not the right one

File: Game/Scripts/CantinaNPCAnimationManager.cpp (hunk 11 above)
Comment (MarcGonzalezMoratona) [inline Game/Scripts/CantinaNPCAnimationManager.cpp:22]:
```suggestion
		LOG_WARNING("{} does not have a Component Audio Source", owner->GetName());
```

File: Game/Scripts/ActivationLogic.cpp [hunk 29]
```diff
@@ -36,10 +36,12 @@ void ActivationLogic::Start()
...
+			LOG_WARNING("{} has not Component Rigid Body", child->GetName());
```
Comment (MarcGonzalezMoratona) [inline Game/Scripts/ActivationLogic.cpp:44]:
```suggestion
			LOG_WARNING("{} does not have Component Rigid Body", child->GetName());
```

File: Game/Scripts/HealthSystem.cpp [hunk 30]
```diff
@@ -81,11 +88,16 @@ void HealthSystem::TakeDamage(float damage)
...
-		{
-			componentParticleSystem->Play();
+		try {
+			if (componentParticleSystem)
+			{
+				componentParticleSystem->Play();
+			}
+		}
+		catch (const ComponentNotFoundException&) {
+			LOG_WARNING("{} not have ComponentParticleSystem", owner->GetName());
```
Comment (MarcGonzalezMoratona) [inline Game/Scripts/HealthSystem.cpp:98]:
```suggestion
			LOG_WARNING("{} does not have ComponentParticleSystem", owner->GetName());
```

File: Game/Scripts/TriggerSewersMusicExit.cpp [hunk 31]
```diff
@@ -26,12 +27,16 @@ void TriggerSewersMusicExit::Start()
...
+		other->GetOwner()->GetComponent<ComponentPlayer>();
 		triggerSewersMusicScript->isMusicTriggered = false;
 
 		AK::SoundEngine::SetState(AUDIO::STATES::GROUP::ZONE, AUDIO::STATES::ID::ZONE::CANTINA);
 		AK::SoundEngine::SetState(AUDIO::STATES::GROUP::LIFE, AUDIO::STATES::ID::PLAYERLIFE::ALIVE);
 		componentAudio->SetSwitch(AUDIO::MUSIC::SWITCH::GROUP::GAMEPLAY, AUDIO::MUSIC::SWITCH::ID::GAMEPLAY::EXPLORATION);
 	}
+	catch (const ComponentNotFoundException&)
+	{
+		LOG_WARNING("{} have not ComponentPlayer", other->GetOwner()->GetName());
```
Comment (MarcGonzalezMoratona) [inline Game/Scripts/TriggerSewersMusicExit.cpp:40]:
```suggestion
		LOG_WARNING("{} does not have ComponentPlayer", other->GetOwner()->GetName());
```

File: Game/Scripts/TriggerSewersMusicExit.cpp (hunk 4 above)
Comment (MarcGonzalezMoratona) [reply Game/Scripts/TriggerSewersMusicExit.cpp:31]: Prever using CompareTag method as se saw in other comments

File: Game/Scripts/UIMissionTrigger.cpp (hunk 5 above)
Comment (MarcGonzalezMoratona) [reply Game/Scripts/UIMissionTrigger.cpp:69]: Prefer using CompareTag method

File: Game/Scripts/UIMissionTrigger.cpp [hunk 32]
```diff
@@ -80,13 +87,14 @@ void UIMissionTrigger::OnCollisionEnter(ComponentRigidBody* other)
 			wasInside = true;
 		}
 	}
+	catch (const ComponentNotFoundException&)
+	{
+		LOG_WARNING("{} have not ComponentPlayer", other->GetOwner()->GetName());
```
Comment (MarcGonzalezMoratona) [inline Game/Scripts/UIMissionTrigger.cpp:92]:
```suggestion
		LOG_WARNING("{} does not have ComponentPlayer", other->GetOwner()->GetName());
```

File: Game/Scripts/UIMissionTrigger.cpp [hunk 33]
```diff
@@ -30,7 +30,14 @@ UIMissionTrigger::~UIMissionTrigger()
 
 void UIMissionTrigger::Start()
 {
-	player = App->GetModule<ModulePlayer>()->GetPlayer()->GetComponent<ComponentPlayer>();
+	try {
+		player = App->GetModule<ModulePlayer>()->GetPlayer()->GetComponent<ComponentPlayer>();
+	}
+	catch (const ComponentNotFoundException&)
+	{
+		LOG_WARNING("{} have not ComponentPlayer", owner->GetName());
```
Comment (MarcGonzalezMoratona) [inline Game/Scripts/UIMissionTrigger.cpp:38]:
```suggestion
		LOG_WARNING("{} does not have ComponentPlayer", owner->GetName());
```

File: Game/Scripts/UITrigger.cpp (hunk 6 above)
Comment (MarcGonzalezMoratona) [reply Game/Scripts/UITrigger.cpp:94]: Prefer using CompareTag method

File: Game/Scripts/UITrigger.cpp [hunk 34]
```diff
@@ -89,16 +90,24 @@ void UITrigger::Update(float deltaTime)
...
+		other->GetOwner()->GetComponent<ComponentPlayer>();
 		onTriggerState = false;
 	}
+	catch (const ComponentNotFoundException&)
+	{
+		LOG_WARNING("{} have not ComponentPlayer", other->GetOwner()->GetName());
```
Comment (MarcGonzalezMoratona) [inline Game/Scripts/UITrigger.cpp:111]:
```suggestion
		LOG_WARNING("{} does not have ComponentPlayer", other->GetOwner()->GetName());
```

File: Game/Scripts/UITrigger.cpp (hunk 7 above)
Comment (MarcGonzalezMoratona) [reply Game/Scripts/UITrigger.cpp:106]: Prefer using CompareTag method

**Comment by MarcGonzalezMoratona** [comment]
Thank you for your work, Eric! Keep in mind the comments about the assets and the try catch blocks

**Comment by gerard-queralt** [comment]
Things to note after my changes:
- Left `try...catch` block for `ComponentParticleSystem` in `HealthSystem`. Doesn't look like a state machine exists for that animation, judging by the names. Updated log message to reflect this is just a patch and needs to be addressed.
- @Dedado3000 says assets are fine to be merged, but I agree with @MarcGonzalezMoratona that some of them don't look correct. Pending confirmation from Level Design team.

**Comment by AuxPort66** [comment]
> Things to note after my changes:
> * @Dedado3000 says assets are fine to be merged, but I agree with @MarcGonzalezMoratona that some of them don't look correct. Pending confirmation from Level Design team.

This was discussed in the meeting and its solved and checked.

**Review by markmeliaa — DISMISSED** [review]
Thanks for taking the time to fix this pr @gerard-queralt, everything looks correctly!

File: Game/Scripts/CantinaNPCAnimationManager.cpp [hunk 35]
```diff
@@ -1,6 +1,7 @@
 #include "CantinaNPCAnimationManager.h"
 #include "Components/ComponentAnimation.h"
 #include "Components/ComponentAudioSource.h"
+#include "AxoLog.h"
```
Comment (MarcGonzalezMoratona) [inline Game/Scripts/CantinaNPCAnimationManager.cpp:4]: I think we don't need this include right now

File: Game/Scripts/TriggerSewersMusic.cpp [hunk 36]
```diff
@@ -1,5 +1,6 @@
 #include "TriggerSewersMusic.h"
 
+#include "AxoLog.h"
```
Comment (MarcGonzalezMoratona) [inline Game/Scripts/TriggerSewersMusic.cpp:3]: do we need this include?

File: Game/Scripts/TriggerSewersMusicExit.cpp [hunk 37]
```diff
@@ -1,5 +1,6 @@
 #include "TriggerSewersMusicExit.h"
 
+#include "AxoLog.h"
```
Comment (MarcGonzalezMoratona) [inline Game/Scripts/TriggerSewersMusicExit.cpp:3]: maybe we don't need this

File: Game/Scripts/UITrigger.cpp [hunk 38]
```diff
@@ -1,4 +1,5 @@
 #include "UITrigger.h"
+#include "AxoLog.h"
```
Comment (MarcGonzalezMoratona) [inline Game/Scripts/UITrigger.cpp:2]: do we need this include?

**Review by MarcGonzalezMoratona — APPROVED** [review]

**Comment by gerard-queralt** [comment]
I've cleaned up the includes, but the network in this camping is so bad I can't pull to update my local branch, so I can't push the commit. Here's the patch, for whoever wants to [apply](https://git-scm.com/docs/git-apply) it xD
[0001-perf-cleaned-up-AxoLog-includes.patch](https://github.com/Horizons-Games/Axolotl-Engine/files/12267967/0001-perf-cleaned-up-AxoLog-includes.patch)

**Review by gerard-queralt — APPROVED** [review]

**Review by MarcGonzalezMoratona — APPROVED** [review]

**Pull request merged**